tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -424,6 +424,204 @@ bool IsDefaultBranch(string candidate) => !StringUtil.IsNullOrWhiteSpace(default
+                    var fetchArgs =
+                        $"fetch --update-shallow --filter=\"blob:none\" --recurse-submodules=no --no-write-fetch-head {remoteName} {commitSha}";
+                    var fetchOutput = RunGitCommand(
+                        workingDirectory,
+                        fetchArgs,
+                        MetricTags.CIVisibilityCommands.Fetch);
+
+                    if (fetchOutput is null || fetchOutput.ExitCode != 0)
+                    {
+                        Log.Warning("GitCommandHelper.FetchCommitData: git fetch failed. Exit={ExitCode}, Error={Error}", fetchOutput?.ExitCode, fetchOutput?.Error);
+                        return null;
+                    }
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            // 3. Get commit details via `git show`
+            Log.Debug("GitCommandHelper.FetchCommitData: fetching commit details for {Commit}", commitSha);
+            var showArgs = $"""show {commitSha} -s --format='%H|,|%at|,|%an|,|%ae|,|%ct|,|%cn|,|%ce|,|%B'""";
+            var showOutput = RunGitCommand(
+                workingDirectory,
+                showArgs,
+                MetricTags.CIVisibilityCommands.GetHead);
+
+            if (showOutput is null || showOutput.ExitCode != 0 || string.IsNullOrWhiteSpace(showOutput.Output))
+            {
+                Log.Warning("GitCommandHelper.FetchCommitData: git show failed. Exit={ExitCode}, Error={Error}", showOutput?.ExitCode, showOutput?.Error);
+                return null;
+            }
+
+            // 4. Parse output
+            // The delimiter is "," (double‑quote, comma, double‑quote)
+            var gitLogDataArray = showOutput.Output.Trim().Split(["|,|"], StringSplitOptions.None);
+            if (gitLogDataArray.Length < 8)
+            {
+                Log.Warning<int>("GitCommandHelper.FetchCommitData: unexpected git show output – expected ≥ 8 tokens, got {Count}", gitLogDataArray.Length);
+                return null;
+            }
+
+            // Parse author and committer dates from Unix timestamp
+            if (!long.TryParse(gitLogDataArray[1], out var authorUnixDate))
+            {
+                Log.Warning("Error parsing author date from git log output");
+                return null;
+            }
+
+            if (!long.TryParse(gitLogDataArray[4], out var committerUnixDate))
+            {
+                Log.Warning("Error parsing committer date from git log output");
+                return null;
+            }
+
+            var commit = gitLogDataArray[0];
+            if (commit.StartsWith("'"))
+            {
+                commit = commit.Substring(1);
+            }
+
+            var commitMessage = string.Join("|,|", gitLogDataArray.Skip(7)).Trim();
+            if (commitMessage.EndsWith("'"))
+            {
+                commitMessage = commitMessage.Substring(0, commitMessage.Length - 1).Trim();
```

Comment:
nit: There's a _lot_ of substringing and trimming going on. You might want to consider working with `ReadOnlySpan<T>` to avoid all the allocations, and just do a final `ToString()` at the end